cpp/cgi_progs/translator_statistics.cc

```diff
@@ -0,0 +1,209 @@
+#include "IniFile.h"
+#include "StringUtil.h"
+#include "Template.h"
+#include "UBTools.h"
+#include "UrlUtil.h"
+#include "util.h"
+#include "WebUtil.h"
+
+
+namespace {
+
+const std::string CONF_FILE_PATH(UBTools::GetTuelibPath() + "translations.conf");
+
+enum Category { VUFIND, KEYWORDS };
+
+
+DbResultSet ExecSqlAndReturnResultsOrDie(const std::string &select_statement, DbConnection * const db_connection) {
+    db_connection->queryOrDie(select_statement);
+    return db_connection->getLastResultSet();
+}
+
+
+// std::vector<std::string> GetLanguageCodesFromTable(DbConnection &db_connection, const std::string &table_name) {
+//     const std::string query("SELECT DISTINCT language_code from " + table_name + " ORDER BY language_code;");
+//     DbResultSet result_set(ExecSqlAndReturnResultsOrDie(query, &db_connection));
+//     std::vector <std::string> language_codes;
+//     while (const DbRow db_row = result_set.getNextRow())
+//         language_codes.emplace_back(db_row["language_code"]);
+
+//     return language_codes;
+// }
+
+
+// std::vector<std::string> GetLanguageCodes(DbConnection &db_connection) {
+//     std::vector <std::string> language_codes(GetLanguageCodesFromTable(db_connection, "vufind_translations"));
+//     for (auto &language_code : GetLanguageCodesFromTable(db_connection, "keyword_translations")) {
+//         if (std::find(language_codes.begin(), language_codes.end(), language_code) == language_codes.end())
+//             language_codes.emplace_back(language_code);
+//     }
+//     return language_codes;
+// }
+
+
+// void ShowErrorPageAndDie(const std::string &title, const std::string &error_message, const std::string &description = "") {
+//     std::cout << "Content-Type: text/html; charset=utf-8\r\n\r\n";
+//     std::cout << "<!DOCTYPE html><html><head><title>" + title + "</title></head>"
+//               << "<body>"
+//               << "  <h1>" + error_message + "</h1>"
+//               << "  <h3>" + description + "</h3>"
+//               << "</body>"
+//               << "</html>";
+//     std::exit(EXIT_SUCCESS);
+// }
+
+
+
+std::string GetCGIParameterOrDefault(const std::multimap<std::string, std::string> &cgi_args,
+                                     const std::string &parameter_name,
+                                     const std::string &default_value) {
+    const auto key_and_value(cgi_args.find(parameter_name));
+    if (key_and_value == cgi_args.cend())
+        return default_value;
+
+    return key_and_value->second;
+}
+
+void GetVuFindStatisticsAsHTMLRowsFromDatabase(DbConnection &db_connection, std::vector<std::string> * const rows, const std::string &start_date, const std::string &end_date)
+{
+    std::string query("select language_code,count(distinct token) as number from vufind_translations where next_version_id is null and prev_version_id is not null and create_timestamp >= '" + start_date + "' and create_timestamp <= '" + end_date + "' group by language_code;");
```

Comment:
It would be nice to have SQL keywords like SELECT, WHERE, AND.... in uppercase letters